COVID Testing sites map blog post (#1575)
agrc/gis.utah.gov: _posts/2020-11-10-covid-testing-sites-map-resources-relationships.md

![The usage graph mirrors the surge in cases over the summer]({% link images/covid-testing-summer.png %})
{: .flex .flex--center}

We saw a rise in usage over the summer as Utah experienced a surge in cases after Memorial Day. I began to feel like the map was living up to its promise. People were getting tested, and because of this testing they would know if they needed to take individual steps to stop the spread. We continued to add sites as new providers came online and make modifications to existing sites. Many sites shifted their hours towards the morning to avoid the summer heat.

![The usage in fall makes all earlier usage seem insignificant]({% link images/covid-testing-surge.png %})
{: .flex .flex--center}

However, as fall progressed we faced the grim reality that we're not out of the woods yet. The usage rose again alongside active cases. The last three weeks in particular have seen massive jumps. The summer usage bump that used to seem so significant almost disappears now as the usage climbs ever upward. The largest single day swing came after the Governor announced new plans to try to stop the spread on November 6th.

> As I've watched this, I've been painfully aware that our usage graph is just a proxy variable for active cases, hospitalizations, and, ultimately, deaths. It's a proxy for overworked nursers and doctors, concerned parents, and grieving family and friends.

Author: ```suggestion
As I've watched this, I've been painfully aware that our usage graph is just a proxy variable for active cases, hospitalizations, and, ultimately, deaths. It's a proxy for overworked nurses and doctors, concerned parents, and grieving family and friends.
```

Since launching the map, we've worked with the Dept. of Health to update both the underlying data and the structure of the map itself. We work in tandem to get data from providers and update it on the website, aided by a shared tracking spreadsheet. When we spun up a Spanish version of the map, they already had someone to help with translating everything, and this relationship has continued as each site's instructions are translated as needed. This relationship allows us to respond to requests quickly and efficiently.

### External Relationships

Perhaps the most critical relationships we've witnessed have been between the Dept. of Health and the individual healthcare providers, especially in rural areas. We've gotten just a glimpse of how these relationships, built over years of visits, calls, and emails, have smoothed the process of getting information to and from providers who may not have the resources of a large hospital network. These bonds not always built easily but are vital to the map's statewide accuracy.

> Then there are our relationships with larger, broader organizations. We've worked with Twenty Labs, the contractors who built the HealthyTogether app for the State and other institutions, to ensure data flows from our map to theirs easily. One of the challenges we've faced is sharing data our data with larger outside groups, like GISCorps or Google, while still keeping it in a format internally that serves the needs of our primary users. This is still a work in progress.

Reviewer: ```suggestion
Then there are our relationships with larger, broader organizations. We've worked with Twenty Labs, the contractors who built the HealthyTogether app for the State and other institutions, to ensure data flows from our map to theirs easily. One of the challenges we've faced is sharing our data with larger outside groups, like GISCorps or Google, while still keeping it in a format internally that serves the needs of our primary users. This is still a work in progress.
```

### Hosted Feature Services

For our [data source](https://utah.maps.arcgis.com/home/item.html?id=83b9c8ea89074aedbaea17cfbde98dec), we started with a copy of the nation-wide, volunteer-driven dataset being created by [GISCorps](https://covid-19-giscorps.hub.arcgis.com/pages/contribute-covid-19-testing-sites-data). We created our own copy, instead of updating the existing dataset directly, to allow us to manage the schema according to our needs and have more direct control over what sites are shown.

Because our data source is a [hosted feature layer](https://doc.arcgis.com/en/arcgis-online/reference/feature-layers.htm) in ArcGIS Online, we can use a single, unified dataset for multiple purposes. We have a private view for editing that allows either us or the Dept. of Health to update the data or add new sites. We also have the public version (linked at the beginning of this section) used by the map and publicly available to anyone else who wants it. We don't have to worry about changes propagating through different copies.

> As a feature service, our data set can also be accessed programmatically via its [REST](https://developers.arcgis.com/rest/) endpoint. This enabled Twenty Labs to scrape our data on a regular interval for their app, and has also allowed the coronavirus.utah.gov website folks from the Dept. of Technology Services  to create a [list-based view](https://coronavirus.utah.gov/covid-testing-locations-list) of the testing site data. They were able to put this interactive app together in just a few days because they can call the REST API directly with a simple HTTP request. They didn't have to learn any new frameworks, and they don't have to worry about maintaining their own version of the data.

Reviewer: **nitpick**: remove double space
```suggestion
As a feature service, our data set can also be accessed programmatically via its [REST](https://developers.arcgis.com/rest/) endpoint. This enabled Twenty Labs to scrape our data on a regular interval for their app, and has also allowed the coronavirus.utah.gov website folks from the Dept. of Technology Services to create a [list-based view](https://coronavirus.utah.gov/covid-testing-locations-list) of the testing site data. They were able to put this interactive app together in just a few days because they can call the REST API directly with a simple HTTP request. They didn't have to learn any new frameworks, and they don't have to worry about maintaining their own version of the data.
```

> As a feature service, our data set can also be accessed programmatically via it's [REST](https://developers.arcgis.com/rest/) endpoint. This enabled Twenty Labs to scrape our data on a regular interval for their app, and has also allowed the coronavirus.utah.gov website folks from the Dept. of Technology Services  to create a [list-based view](https://coronavirus.utah.gov/covid-testing-locations-list) of the testing site data. They were able to put this interactive app together in just a few days because they can call the REST API directly with a simple HTTP request. They didn't have to learn any new frameworks, and they don't have to worry about maintaining their own version of the data.

Reviewer 2: ```suggestion
As a feature service, our data set can also be accessed programmatically via its [REST](https://developers.arcgis.com/rest/) endpoint. This enabled Twenty Labs to scrape our data on a regular interval for their app, and has also allowed the coronavirus.utah.gov website folks from the Dept. of Technology Services  to create a [list-based view](https://coronavirus.utah.gov/covid-testing-locations-list) of the testing site data. They were able to put this interactive app together in just a few days because they can call the REST API directly with a simple HTTP request. They didn't have to learn any new frameworks, and they don't have to worry about maintaining their own version of the data.
```

### Scaling

> ArcGIS Online has scaled seamlessly as our usage has taken off. Despite being relatively simple and only being seven months old, our dataset already has more requests in the last 12 months than any of our other ArcGIS Online-hosted datasets. The map has nearly 800,000 hits and is now averaging 10,000+ new hits a day. If we were hosting this on our own hardware with a custom application, we would be scrambling to upgrade the VM, feed it more bandwidth, and fix coding problems that matter when you've got thousands of hits.

Author: ```suggestion
ArcGIS Online has scaled seamlessly as our usage has taken off. Despite being relatively simple and only being seven months old, our dataset already has more requests in the last 12 months than any of our other ArcGIS Online-hosted datasets. The map has nearly 800,000 hits and is now averaging 10,000+ new hits a day. Thanks to ArcGIS Online, we don't have to worry about monitoring server performance or bandwidth usage.
```

> ArcGIS Online has scaled seamlessly as our usage has taken off. Despite being relatively simple and only being seven months old, our dataset already has more requests in the last 12 months than any of our other ArcGIS Online-hosted datasets. The map has nearly 800,000 hits and is now averaging 10,000+ new hits a day. If we were hosting this on our own hardware with a custom application, we would be scrambling to upgrade the VM, feed it more bandwidth, and fix coding problems that matter when you've got thousands of hits.

Reviewer: **suggestion**: give us a little credit. we can deal with that traffic.
```suggestion
ArcGIS Online has scaled seamlessly as our usage has taken off. Despite being relatively simple and only being seven months old, our dataset already has more requests in the last 12 months than any of our other ArcGIS Online-hosted datasets. The map has nearly 800,000 hits and is now averaging 10,000+ new hits a day.
```

Author: Fair enough, I suppose my brain is still in the pre-VM days where a single app taking off meant your normally small and quiet GIS server buckled under the load of running an app you cobbled together in a hurry with plenty of technical debt. 

What about "Thanks to ArcGIS Online, we don't have to worry about monitoring server performance or bandwidth."?

Reviewer: yah that is much more positive language
